xlint

```diff
@@ -123,6 +123,22 @@ file_end() {
 	fi
 }
 
+revision_reset() {
+	[ "${revision}" = 1 ] && return
+	# inside git repo?
+	git -C "$(dirname $template)" rev-parse HEAD >/dev/null 2>/dev/null || return
+	remote=${XLINT_UPSTREAM:-$(git -C "$(dirname $template)" remote -v | grep -P [:/]void-linux/void-packages | head -n1 | cut -f1)}
+	prev_version=$(git -C "$(dirname $template)" show ${remote}/master:./$(basename $template) 2>/dev/null | grep -Po '^version=\K.*')
+	xbps-uhelper cmpver "$version" "$prev_version"
+	[ $? = 1 ] || return
+	if [ "$prev_version" ]; then
+		action=update
+	else
+		action=creation
+	fi
```

Comment:
I don't really like to use porcelain command inside script.
I think `git-show`'s interface is stable enough.
Probably better to use `git diff --word-diff=porcelain`?

Reply:
Changed to cat-file blob.
There is also usage of `git remote` here, to make it just work locally. lint-commits use FETCH_HEAD instead, but that points to PR head and not base if `git fetch pull/#/head` was called before, what is not what we want.